pkg/scheduler/framework/plugins/noderesources/balanced_allocation.go

```diff
@@ -67,42 +68,67 @@ func (ba *BalancedAllocation) ScoreExtensions() framework.ScoreExtensions {
 }
 
 // NewBalancedAllocation initializes a new plugin and returns it.
-func NewBalancedAllocation(_ runtime.Object, h framework.Handle, fts feature.Features) (framework.Plugin, error) {
+func NewBalancedAllocation(baArgs runtime.Object, h framework.Handle, fts feature.Features) (framework.Plugin, error) {
+	args, ok := baArgs.(*config.NodeResourcesBalancedAllocationArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type NodeResourcesBalancedAllocationArgs, got %T", baArgs)
+	}
+
+	if err := validation.ValidateNodeResourcesBalancedAllocationArgs(nil, args); err != nil {
+		return nil, err
+	}
+
+	resToWeightMap := make(resourceToWeightMap)
+	for _, resource := range args.Resources {
+		resToWeightMap[v1.ResourceName(resource.Name)] = 1
+	}
+
 	return &BalancedAllocation{
 		handle: h,
 		resourceAllocationScorer: resourceAllocationScorer{
 			Name:                BalancedAllocationName,
 			scorer:              balancedResourceScorer,
-			resourceToWeightMap: defaultRequestedRatioResources,
+			resourceToWeightMap: resToWeightMap,
 			enablePodOverhead:   fts.EnablePodOverhead,
 		},
 	}, nil
 }
 
-// todo: use resource weights in the scorer function
 func balancedResourceScorer(requested, allocable resourceToValueMap) int64 {
-	cpuFraction := fractionOfCapacity(requested[v1.ResourceCPU], allocable[v1.ResourceCPU])
-	memoryFraction := fractionOfCapacity(requested[v1.ResourceMemory], allocable[v1.ResourceMemory])
-	// fractions might be greater than 1 because pods with no requests get minimum
-	// values.
-	if cpuFraction > 1 {
-		cpuFraction = 1
-	}
-	if memoryFraction > 1 {
-		memoryFraction = 1
+	resourceToFractions := []float64{}
+	var totalFraction float64
+	for name, value := range requested {
+		// This could only happen if both requsted[resource] and allocable[resource] are zero, this case could be avoided by tainting
+		// the node so that the node with the scare resources could be untouched and leave more room for the pods which do need them.
```

Comment:
the comment should say "this should never happen because the filter wouldn't have passed this node in the first place"

Reply:
actually, this could happen if both the `requested` and `allocable` are zero, have created a testcase to cover this: "node without the scare resource results to a higher score", pls see line @364 in this file `balanced_allocation_test.go`

Reply:
those line has been removed.